perlock-core/src/main/java/name/mitterdorfer/perlock/SinglePathWatcherFactory.java

```diff
@@ -0,0 +1,107 @@
+package name.mitterdorfer.perlock;
+
+import name.mitterdorfer.perlock.impl.util.Preconditions;
+
+import java.nio.file.Path;
+
+import java.nio.file.WatchEvent;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static java.nio.file.StandardWatchEventKinds.*;
+
+/**
+ * Factory to create {@link PathWatcher}s that only deal with a single file and
+ * ignore change to other files within the directory
+ */
+public class SinglePathWatcherFactory {
```

Comment:
Also, `isRegularFile` will return false if the file does not exist.
The user will not be able to watch for file creation unless they create and delete the file before and after starting the path watcher.